doc/migrating_to_spire_agent_id_enforced_format.md

```diff
@@ -0,0 +1,101 @@
+# SPIRE Agent ID Migration Guide
+
+This guide outlines the steps required to update your SPIRE agent IDs to conform to the enforced ID shape.
+
+<!-- todo(matteus): mention first version which enforced the shape -->
+
+Before vX.Y.Z, SPIRE did not enforce a specific format for agent IDs. However, for improved debuggability, auditability, and security, it was recommended to follow this format:
+
+```
+spiffe://<trustdomain>/spire/agent/<nodeattestor>/<unique per-agent suffix>
+Example:
+```
+
+Now SPIRE enforces that agent IDs reside in the agent namespace (`spiffe://<trustdomain>/spire/agent`), like in the following concrete example:
+
+```
+spiffe://example.org/spire/agent/join_token/21B6D625-CCF3-49E1-8E7C-812B3F55B3CB
+```
+
+
+<!-- ------------------------------------------------------------------- -->
+
+## Why this Update?
+
+Following the recommended format provides several benefits:
+
+1. Easier Debugging and Auditing: The format clearly identifies the node attestor that generated the ID and aids in log analysis.
+
+2. Reduced Misconfiguration Risk: Validating requested IDs against the format helps prevent accidental workload assignment of agent IDs.
+
+3. Improved Clarity: The format provides a clear picture of the ID's origin at a glance.
+
+
+## Impact
+
+SPIRE enforces the recommended format for agent IDs starting with version vX.Y.Z. 
+
+<!-- todo(matteus): mention first version which enforced the shape -->
+<!-- todo(matteus): is the following plan still valid?
+
+From Andrew:
+-------------------
+Warn on the undesired ID usage (this is already merged and will ship in 1.2.1. Since this was not in place for 1.2.0, we cannot change it through 1.3.x)
+
+-> Beginning with 1.4.0 (THIS HAS CHANGED, RIGHT?), disallow newly attested nodes which do not conform to the expected ID shape. Existing agents will still continue to operate successfully. As part of this change, we will also introduce a warning for existing IDs that are non-conformant.
```

Comment:
Yes, this has very much changed :) We introduced the warning in 1.2.1. We decided that we couldn't start enforcing until this migration guide was written. Once this guide has been authored and merged, the next minor release of SPIRE will start enforcing new agents to have that ID shape. The following minor version will start rejecting old agents.